### ISSUE TITLE
Option "/MANIFEST:NO" in CMakeLists.txt for the MSVC kit

### ISSUE THREAD

**Issue by jgueytat** [KDAB/GammaRay#458]
MSVC2015 32bit / QtCreator / Qt 5.9.1 / Tag 2.8.1
With the MSVC kit under windows the compilation fails if you don't add the option "/MANIFEST:NO" to the linker. If you are not able to disable your antivirus for the build directory you need to add this option to the linker.
It would be nice to have access to this option nicely on the CMakeLists.txt.

**Comment by krf** [comment]
Could you try invoking cmake as follows?
```
cmake -DCMAKE_EXE_LINKER_FLAGS="/manifest:no" -DCMAKE_MODULE_LINKER_FLAGS="/manifest:no" CMAKE_SHARED_LINKER_FLAGS="/manifest:no" ...
```

**Comment by krf** [comment]
I don't think that's something GammaRay needs to fix in its build system, that's rather a setup problem (which you're the first to report, as far as I can see)

**Comment by jgueytat** [comment]
The compilation does work with those following lines : 

- set(CMAKE_EXE_LINKER_FLAGS    "${CMAKE_EXE_LINKER_FLAGS} /MANIFEST:NO")
- set(CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} /MANIFEST:NO")
- set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} /MANIFEST:NO")

So your CMake invokation would work also.

**Comment by jgueytat** [comment]
May be it's worth a warning somewhere in the documentation while compiling with MSVC otherwise.

**Comment by krf** [comment]
Could you tell us in what ways the compilation fails (terminal output) so we can also reference this?

**Comment by jgueytat** [comment]
Not able to reproduce it.
I got instead only some warnings related to MT now.

```
Creating library gammaray_3dinspector_uid.lib and object gammaray_3dinspector_uid.exp
MT: command "mt /nologo /manifest CMakeFiles\gammaray_3dinspector_ui.dir/intermediate.manifest /out:CMakeFiles\gammaray_3dinspector_ui.dir/embed.manifest /notify_update" failed (exit code 0x41020001) with the following output:
   Creating library gammaray_3dinspector_uid.lib and object gammaray_3dinspector_uid.exp
```

**Comment by jgueytat** [comment]
The previous error I got was.

`mt.exe : general error c101008d: Failed to write the updated manifest to the resource of file … Access is denied`

You can find explanations [on StackOverflow](https://stackoverflow.com/questions/3775406/mt-exe-general-error-c101008d-failed-to-write-the-updated-manifest-to-the-res).

**Comment by krf** [comment]
Can you confirm the solution from https://github.com/KDAB/GammaRay/issues/458#issuecomment-337827457 works?

**Comment by gunterkoenigsmann** [comment]
I know someone with this kind of braindead virus scanner which causes the Following to happen:

- The C Compiler creates the application
- Once this step is finished the Virus Scanner scans the file and while the file is being scanned it cannot be modified.
- Then a tool tries to modify the application's embedded manifest.

Another way to trigger a similar problem is to shut down the "Enhance the user experience for applications" service.

I personally would judge this to be a bug in the Operating System or the toolchain and not in CMake => I agree with closing this bug.
